### DISCUSSION TITLE
Enable AutoMerge on Repo with CODEOWNERS

### DISCUSSION THREAD

**Discussion by duboisford** [renovatebot/renovate#18328] [Request Help] [answered]
Hello. I'm a newbie to renovate and am working on an internal gitlab system. Renovate successfully generates merge requests for the repo I'm working with and I can see in the MR the traffic light with "Automerge: Enabled" next to it. However, this repo has a setting that requires one approval from CODEOWNERS before an MR can be merged. I tried adding `service-renovate` to the CODEOWNERS and adding a few config settings, specifically  `"gitLabIgnoreApprovals": true` under packageRules that I thought would handle this. Is what I am trying to do possible? How can I get MR's generated by service-renovate approved in an automated manner?

``` 
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "matchUpdateTypes": ["major", "minor", "patch", "pin", "digest"],
      "automerge": true,
      "platformAutomerge": true,
      "gitLabIgnoreApprovals": true
    }
  ]
}
```

**Comment by viceice** [comment, marked as answer]
renovate can't and won't approve it's own PR

**Reply by duboisford** [reply]
Thanks for the quick reply. Do you know of any work-arounds or have any ideas on other ways I can accomplish this?

**Reply by viceice** [reply]
use a second separate bot to approve, like the Renovate approve bots on GitHub

**Reply by HonkingGoose** [reply]
> use a second separate bot to approve, like the Renovate approve bots on GitHub

Our docs say the approve bots are for **GitHub** only. They are on **GitLab**. Quote from [Renovate docs, Automerge -> Required Pull Request reviews](https://docs.renovatebot.com/key-concepts/automerge/#required-pull-request-reviews):

> 
> ### Required Pull Request reviews
> 
> If you have mandatory Pull Request reviews then it means Renovate can't automerge its own PR until such a review has happened.
> 
> If you're on `github.com` or GitHub Enterprise Server (`>=3.4`) you can let Renovate bypass the mandatory Pull Request reviews using the "[Allow specified actors to bypass required pull requests](https://github.blog/changelog/2021-11-19-allow-bypassing-required-pull-requests/)" option in your branch protection rules.
> 
> Alternatively, if you are running the hosted Mend Renovate App on `github.com`, you can also install the helper apps [renovate-approve](https://github.com/apps/renovate-approve) and [renovate-approve-2](https://github.com/apps/renovate-approve-2) and they will mark all automerging Pull Requests by Renovate as approved.
> These approval helper apps are only available for GitHub.

**Reply by duboisford** [reply]
Thanks. I'll ask my org to install the renovate-approve app on our system.
